integration_test/Makefile

```diff
@@ -211,3 +211,30 @@ pg13:
 	fi
 	jq '.queryInformations[].normalizedQuery' pg13_2.json | sort --version-sort -f > pg13.out
 	diff -Naur pg13.expected pg13.out && echo 'success'
+
+guided-setup:
+	docker build -f Dockerfile.test-guided-setup $(DOCKER_BUILD_OPTS)
+	$(DOCKER_RUN_CMD)
+	sleep 10
+	docker exec --user postgres pganalyze-collector-test pg_ctl --pgdata /var/lib/postgresql/10/main --wait start --log /tmp/postgres.log --options "-c config_file=/etc/postgresql/10/main/postgresql.conf"
+	docker exec --user pganalyze --detach pganalyze-collector-test pganalyze-collector
+
+	docker exec --privileged pganalyze-collector-test env PGA_SETUP_USE_PG_CTL=true pganalyze-collector-setup --api-key=abc123 --recommended --db-name=postgres
+
+	docker exec --user postgres pganalyze-collector-test psql -c "SHOW shared_preload_libraries" \
+	  -c "SELECT count(*) FROM pg_settings WHERE pending_restart" \
+	  -c "SELECT name, setting FROM pg_settings WHERE name IN ('log_duration', 'log_error_verbosity', 'log_line_prefix', 'log_min_duration_statement', 'log_statement', 'auto_explain.log_analyze', 'auto_explain.log_buffers', 'auto_explain.log_timing', 'auto_explain.log_triggers', 'auto_explain.log_verbose', 'auto_explain.log_format', 'auto_explain.log_min_duration', 'auto_explain.log_nested_statements') ORDER BY name" \
+	  -c "SELECT pg_has_role('pganalyze', 'pg_monitor', 'usage')" \
+	  > guided-setup.postgres.out
+	docker exec --user pganalyze pganalyze-collector-test cat /etc/pganalyze-collector.conf > guided-setup.collector-config.out
+	docker exec pganalyze-collector-test sh -c "PGA_DISABLE_ACTIVITY=1 pganalyze-collector --dry-run --force-state-update -v" > guided-setup.snapshot.json.out
+
+	docker exec --user postgres pganalyze-collector-test pg_ctl --pgdata /var/lib/postgresql/10/main --wait stop --mode fast
+	docker rm -f pganalyze-collector-test
+	docker rmi pganalyze-collector-test
+	diff -Nau guided-setup.postgres.expected guided-setup.postgres.out && echo 'postgres configuration modified as expected'
+	grep -v db_password guided-setup.collector-config.out > guided-setup.collector-config.nopass.out
+	diff -Nau guided-setup.collector-config.nopass.expected guided-setup.collector-config.nopass.out \
+	  && grep --extended-regexp --quiet 'db_password\s+=\s+[a-f0-9]{32}' guided-setup.collector-config.out && echo 'collector config file modified as expected'
+	jq '{fns: .functionReferences, roles: .roleReferences[] |  select(.name == "pganalyze")}' < guided-setup.snapshot.json.out > guided-setup.snapshot-subset.json.out
+	diff -Nau guided-setup.snapshot-subset.json.expected guided-setup.snapshot-subset.json.out && echo 'success'
```

Comment:
@lfittl I was thinking about what to verify here, and since we're testing the setup program, I did a `--dry-run` smoke test to ensure that the collector will run, but I mostly wanted to test that Postgres and the config file are actually set up correctly.

One issue I ran into is that the current `pg_ctl restart` mechanism somehow kills Postgres logging, so the log test fails: log output stops when the server shuts down, and does not resume again when the server restarts. I think this has to do with Go's handling of stdout/stderr in `exec.Command`, but I'm not sure how to fix this. I tried throwing away my hacky stdout/stderr mechanism and just install sudo as we'd discussed, but I hit the same problem. The test does still run, but it's not able to validate log setup.

Reply:
Tried `nohup` as well just in case, but no dice. I also tried setting both Stderr and Stdout to `ioutil.Discard`, but also no dice.

Reply:
We may be able to use systemd after all, if it helps us avoid the `pg_ctl` issue here.

I explored that in https://github.com/pganalyze/collector/pull/116/commits/6080a1bc856cf159949ad6cac8f93d77c54b82c0 based on some tricks I found. That seems to work for both Ubuntu 18.04 image (as tested there) and Ubuntu 20.04.

Maybe that is the right approach after all? (note I didn't test the log aspect yet, but does run Postgres using systemd successfully)